### PR TITLE
Tl 286

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/TL-286)
## Describe this PR

Patch endpoint for Updating with new Responsible Entity

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        public GetByIdRequestValidatorTests()
+        {
+            _classUnderTest = new GetByIdRequestValidator();
+        }
+
+        [Fact]
+        public void RequestShouldErrorWithNullId()
+        {
+            var query = new PatchesQueryObject();
```
Comment (evak6979) [inline PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs:24]: Perhaps explicitly set the Id to null

File: PatchesApi.Tests/V1/Boundary/Validation/UpdatePatchQueryValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        public UpdatePatchQueryValidatorTests()
+        {
+            _classUnderTest = new UpdatePatchQueryValidator();
+        }
+
+        [Fact]
+        public void RequestShouldErrorWithNullId()
+        {
+            var query = new UpdatePatchesResponsibilityRequest();
```
Comment (evak6979) [inline PatchesApi.Tests/V1/Boundary/Validation/UpdatePatchQueryValidatorTests.cs:24]: As before. Explicitly set Id to null

File: PatchesApi.Tests/V1/Controllers/PatchesApiControllerTests.cs [hunk 3]
```diff
@@ -91,5 +139,80 @@ public void GetPatchByIdExceptionIsThrown()
             func.Should().Throw<ApplicationException>().WithMessage(exception.Message);
         }
 
+        [Fact]
+        public async Task UpdatePatchByResponsibilityAsyncFoundReturnsFound()
+        {
+            // Arrange
```
Comment (evak6979) [inline PatchesApi.Tests/V1/Controllers/PatchesApiControllerTests.cs:145]: Some of the tests have Arrange/Act, some not. I suggest consistency

File: PatchesApi/V1/Boundary/Request/Validation/ErrorCodes.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatchesApi.V1.Boundary.Request.Validation
+{
+    public static class ErrorCodes
+    {
+        public const string XssCheckFailure = "W42";
```
Comment (evak6979) [inline PatchesApi/V1/Boundary/Request/Validation/ErrorCodes.cs:10]: Should this perhaps be part of the Constants class (if we have one?)

File: PatchesApi/V1/Boundary/Request/Validation/PatchValidator.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        public PatchValidator()
+        {
+            RuleFor(x => x.Id).NotNull()
+                .NotEqual(Guid.Empty);
+            RuleFor(x => x.Name).NotXssString()
+                         .WithErrorCode(ErrorCodes.XssCheckFailure);
+            RuleFor(x => x.Domain).NotXssString()
+                         .WithErrorCode(ErrorCodes.XssCheckFailure);
+            RuleForEach(x => x.ResponsibleEntities).SetValidator(new ResponsibleEntitiesValidator());
+
```
Comment (evak6979) [inline PatchesApi/V1/Boundary/Request/Validation/PatchValidator.cs:22]: Remove extra space

File: PatchesApi/V1/Gateways/PatchesGateway.cs [hunk 6]
```diff
@@ -31,5 +34,34 @@ public async Task<PatchEntity> GetPatchByIdAsync(PatchesQueryObject query)
...
+            if (ifMatch != patch.VersionNumber)
+                throw new VersionNumberConflictException(ifMatch, patch.VersionNumber);
+            var responsibleEntity = new ResponsibleEntities()
+            {
+                Id = query.ResponsibileEntityId,
+                Name = requestObject.Name,
+                ResponsibleType = requestObject.ResponsibleType
+            };
+            patch.ResponsibleEntities.Add(responsibleEntity);
+
```
Comment (evak6979) [inline PatchesApi/V1/Gateways/PatchesGateway.cs:55]: Too much space

File: PatchesApi/V1/UseCase/UpdatePatchResponsibilitiesUseCase.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            _gateway = gateway;
+        }
+
+        [LogCall]
+        public async Task<PatchesResponseObject> ExecuteAsync(UpdatePatchesResponsibilityRequest query, UpdatePatchesResponsibilitiesRequestObject updateRequestObject,
+             int? ifMatch)
+        {
+            var updateResult = await _gateway.UpdatePatchResponsibilities(query, updateRequestObject, ifMatch).ConfigureAwait(false);
+            if (updateResult == null) return null;
+
```
Comment (evak6979) [inline PatchesApi/V1/UseCase/UpdatePatchResponsibilitiesUseCase.cs:30]: Too much space

File: PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs (hunk 1 above)
Comment (humulla) [reply PatchesApi.Tests/V1/Boundary/Validation/GetByIdRequestValidatorTests.cs:24]: I don't think we can do that as Id is a Guid value so I can't set it to null

File: PatchesApi.Tests/V1/Boundary/Validation/UpdatePatchQueryValidatorTests.cs (hunk 2 above)
Comment (humulla) [reply PatchesApi.Tests/V1/Boundary/Validation/UpdatePatchQueryValidatorTests.cs:24]: I don't think we can do that as Id is a Guid value so I can't set it to null

**Review by evak6979 — APPROVED** [review]

**Pull request merged**